polaris/benchmark/_base.py

```diff
@@ -432,72 +470,32 @@ def evaluate(
-        # See also the `hide_targets` parameter in the `Subset` class.
-        test = self.get_train_test_split()[1]
-
-        if not isinstance(test, dict):
-            test = {"test": test}
-
-        y_true = {}
-        for k, test_subset in test.items():
-            with tmp_attribute_change(test_subset, "_hide_targets", False):
-                y_true[k] = test_subset.targets
-
-        if not isinstance(y_pred, dict) or all(k in self.target_cols for k in y_pred):
-            y_pred = {"test": y_pred}
-
-        if not isinstance(y_prob, dict) or all(k in self.target_cols for k in y_prob):
-            y_prob = {"test": y_prob}
-
-        if any(k not in y_pred for k in test.keys()) and any(k not in y_prob for k in test.keys()):
-            raise KeyError(
-                f"Missing keys for at least one of the test sets. Expecting: {sorted(test.keys())}"
-            )
-
-        # Results are saved in a tabular format. For more info, see the BenchmarkResults docs.
-        scores: ResultsType = pd.DataFrame(columns=BenchmarkResults.RESULTS_COLUMNS)
-
-        # For every test set...
-        for test_label, y_true_subset in y_true.items():
-            # For every metric...
-            for metric in self.metrics:
-                if metric.is_multitask:
-                    # Multi-task but with a metric across targets
-                    score = metric(
-                        y_true=y_true_subset, y_pred=y_pred.get(test_label), y_prob=y_prob.get(test_label)
-                    )
-                    scores.loc[len(scores)] = (test_label, "aggregated", metric, score)
-                    continue
-
-                if not isinstance(y_true_subset, dict):
-                    # Single task
-                    score = metric(
-                        y_true=y_true_subset, y_pred=y_pred.get(test_label), y_prob=y_prob.get(test_label)
-                    )
-                    scores.loc[len(scores)] = (
-                        test_label,
-                        self.target_cols[0],
-                        metric,
-                        score,
-                    )
-                    continue
+        # The `evaluate_benchmark` function expects the benchmark labels to be of a certain structure which
+        # depends on the number of tasks and test sets defined for the benchmark. Below, we build the structure
+        # of the benchmark labels based on the aforementioned factors.
+        test = self._get_test_set(hide_targets=False)
+        if isinstance(test, dict):
+            #
+            # For multi-set benchmarks
+            y_true = {}
+            for test_set_name, values in test.items():
+                y_true[test_set_name] = {}
+                if isinstance(values.targets, dict):
+                    #
+                    # For multi-task, multi-set benchmarks
+                    for task_name, values in values.targets.items():
+                        y_true[test_set_name][task_name] = values
```

Comment:
**Suggestion:** I believe this would do the same thing:

```suggestion
                    y_true[test_set_name] = values.targets
```

Reply:
I believe I tried that initially. For multi-task, multi-target benchmarks, `values.targets` held a structure like this:

`{task_name_1: [values...], task_name_2: [values...]}`